Fix mobile view and rework (s)css to use PrimeVue theming or inline PrimeFlex classes

CodeNamedRobin opened the pull request:
Fixes #29

SuperVK committed:
added central theming and central basepage

SuperVK committed:
added central theming and central basepage

SuperVK committed:
Replace most :deep calls/common CSS

SuperVK committed:
Replace most :deep calls/common CSS

SuperVK committed:
Made userview and containers styling containers

SuperVK committed:
Made userview and containers styling containers

CodeNamedRobin committed:
Fixed some leftover merge-conflicts and issues

CodeNamedRobin committed:
Fixed weird padding issue

CodeNamedRobin committed:
Fixed MutationTableComponent layout

- Also fixed HomeView mobile layout

CodeNamedRobin committed:
Added quick bar overview

CodeNamedRobin committed:
Re-added accidentally removed variable

CodeNamedRobin committed:
Some small improvements

- Added translations to TopupModalComponent
- Made ProfileView.vue mobile friendly
- Fixed Saldo link in the TopNavbar
- Made POSOverviewTableComponent mobile friendly
- Fixed Copyright banner sometimes overlapping content
- Made POSCreateView mobile friendly
- Removed all css from POSCreateView and reworked using PrimeFlex

CodeNamedRobin committed:
Further styling improvements

- Reworked most scss into PrimeFlex inline classes in POSEditView.vue

- Made POSEditView.vue mobile friendly
- Removed handleClosed function in POSInfoView
- Reworked ProductGridComponent to use PrimeFlex classes
- ProductGridComponent is now responsive to screen size
- Reworked ProductComponent to use PrimeFlex classes
- Re-ordered some components in POSInfoView
- POSInfoView is now mobile friendly
- Reworked the modal for creating users to be mobile friendly
- Reworked styling of SingleUserView.vue
- Made SingleUserView.vue mobile friendly
- Reworked styling of ProductsContainersView.vue
- Made ProductsContainersView.vue mobile friendly(ish)
- Remade CreateProductDialog and made mobile friendly
- Reworked and made remaining dialogs mobile friendly

CodeNamedRobin committed:
Removed more unnecessary scss

CodeNamedRobin committed:
Replaced defineComponentBinds with defineField in ProductCreateComponent.vue

SuperVK commented:
Bottom bar is not sticky to the bottom of the screen when content doesn't fill the entire screen
![image](https://github.com/GEWIS/sudosos-frontend-vue3/assets/31625323/d3b75b67-eff2-4b8a-b77e-cf19311ffd8f)

SuperVK commented on src/views/BalanceView.vue:
On line 24 there is still a "severity=danger"

SuperVK committed:
Added a little bit of pre-dark theme support

CodeNamedRobin committed:
Removed unnecessary css

CodeNamedRobin committed:
Made improvements according to PR review

- Fixed ProductCreateComponent
- Fixed CopyRightBanner not always staying at the bottom
- Removed one usage of severity="danger" in BalanceView.vue

CodeNamedRobin committed:
Fixed footer to finally stay at bottom

SuperVK commented on src/components/ProductComponent.vue:
```diff
@@ -25,33 +23,33 @@ defineProps({
+//  overflow: hidden;
+//  text-overflow: ellipsis;
+//  white-space: nowrap;
+//}
```
Would be nice to remove all of the commented CSS? Unless there is a good reason for it

SuperVK commented on src/components/ProductDialogComponent.vue:
```diff
@@ -55,41 +72,41 @@ const visible = ref(false);
+//      height: 100%;
+//    }
+//  }
+//}
```
Same goes here

CodeNamedRobin committed:
Removed commented CSS

Some small lint fixes

CodeNamedRobin committed:
Fixed one god-damn type-error

CodeNamedRobin committed:
Made footer sticky

CodeNamedRobin committed:
Fixed ProfileView layout on desktop

CodeNamedRobin committed:
Finally fixed footer

CodeNamedRobin committed:
Fixed mutation modal not being mobile friendly

CodeNamedRobin committed:
Fixed stripe payments

SuperVK commented on src/views/LoginView.vue:
```diff
@@ -152,6 +152,35 @@ const toHomeView = () => {
 
 <style scoped lang="scss">
 //TODO Fix the amount of css used related to #14 and #29
```
This line needs to be removed

SuperVK commented on src/views/PointOfSale/POSEditView.vue:
```diff
@@ -181,81 +178,6 @@ const handleEditPOS = handleSubmit(async (values) => {
 
 <style scoped lang="scss">
 // TODO: Generalize this style and the one from create view
```
This TODO should be fixed with this pull as well right?

CodeNamedRobin replied:
Nah, I'm still thinking about this as I have no clue how to do this efficiently and correctly.

SuperVK commented on src/views/SingleUserView.vue:
```diff
@@ -123,35 +130,33 @@ const getUserMutations = async (take: number, skip: number) :
-  flex-direction: column;
-}
+<style scoped lang="scss">
+//form {
```
Some uncommented code left

SuperVK commented on src/views/TermsOfServiceView.vue:
```diff
@@ -67,7 +67,7 @@ const handleLogout = () => {
 
 <style scoped lang="scss">
-@import '../styles/BasePage.css';
+//@import '../styles/BasePage.css';
```
Same here

SuperVK commented on src/views/TransactionsView.vue:
```diff
@@ -50,5 +50,5 @@ const getUserMutations = async (take: number, skip: number)
 
 <style scoped lang="scss">
-@import '../styles/BasePage.css';
+//@import '../styles/BasePage.css';
```
Same here

CodeNamedRobin committed:
Fixed footer not being on bottom on LoginView

CodeNamedRobin committed:
Removed commented code and reduced some more css

JustSamuel committed:
Fixed + update deps

JustSamuel committed:
Merge remote-tracking branch 'origin/feature/themes' into feature/themes

# Conflicts:
#	package-lock.json

CodeNamedRobin committed:
Fixed type-check errors

CodeNamedRobin committed:
Fixed everything

JustSamuel reviewed (approved):
Mooi
